presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/functionNamespace/NativeFunctionNamespaceManager.java

```diff
@@ -0,0 +1,486 @@
+        }
+
+        Optional<AggregationFunctionMetadata> aggregationFunctionMetadata =
+                jsonBasedUdfFunctionMetaData.getAggregateMetadata()
+                        .map(metadata -> new AggregationFunctionMetadata(
+                                convertApplicableTypeToVariable(metadata.getIntermediateType(), types),
+                                metadata.isOrderSensitive()));
+
+        return new SqlInvokedFunction(
+                qualifiedFunctionName,
+                parameterBuilder.build(),
+                typeVariableConstraintsList,
+                outputType,
+                jsonBasedUdfFunctionMetaData.getDocString(),
+                jsonBasedUdfFunctionMetaData.getRoutineCharacteristics(),
+                "",
+                jsonBasedUdfFunctionMetaData.getVariableArity(),
+                notVersioned(),
+                jsonBasedUdfFunctionMetaData.getFunctionKind(),
+                aggregationFunctionMetadata);
+    }
+
+    @Override
+    protected Collection<SqlInvokedFunction> fetchFunctionsDirect(QualifiedObjectName functionName)
+    {
+        return memoizedFunctionsSupplier.get().values().stream()
+                .filter(function -> function.getSignature().getName().equals(functionName))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    protected UserDefinedType fetchUserDefinedTypeDirect(QualifiedObjectName typeName)
+    {
+        return userDefinedTypes.get(typeName);
+    }
+
+    @Override
+    protected FunctionMetadata fetchFunctionMetadataDirect(SqlFunctionHandle functionHandle)
+    {
+        if (functionHandle instanceof NativeFunctionHandle) {
+            return getMetadataFromNativeFunctionHandle(functionHandle);
+        }
+
+        return fetchFunctionsDirect(functionHandle.getFunctionId().getFunctionName()).stream()
+                .filter(function -> function.getRequiredFunctionHandle().equals(functionHandle))
+                .map(this::sqlInvokedFunctionToMetadata).collect(onlyElement());
+    }
+
+    @Override
+    protected ScalarFunctionImplementation fetchFunctionImplementationDirect(SqlFunctionHandle functionHandle)
+    {
+        return fetchFunctionsDirect(functionHandle.getFunctionId().getFunctionName()).stream()
+                .filter(function -> function.getRequiredFunctionHandle().equals(functionHandle))
+                .map(this::sqlInvokedFunctionToImplementation)
+                .collect(onlyElement());
+    }
+
+    @Override
+    public synchronized void createFunction(SqlInvokedFunction function, boolean replace)
+    {
+        checkFunctionLanguageSupported(function);
+        SqlFunctionId functionId = function.getFunctionId();
+        if (!replace && functions.containsKey(function.getFunctionId())) {
+            throw new PrestoException(GENERIC_USER_ERROR, format("Function '%s' already exists", functionId.getId()));
```

Comment:
I would create a new error code type in StandardErrorCodes.  `DUPLICATE_FUNCTION_ERROR`.